odl/trafos/wavelet.py

```diff
@@ -683,14 +1020,14 @@ def inverse(self):
         """The inverse wavelet transform."""
         return WaveletTransformInverse(
             ran=self.domain, nscales=self.nscales, wbasis=self.wbasis,
-            mode=self.mode)
+            mode=self.mode, axes=self._axes)
 
 
 class WaveletTransformInverse(Operator):
 
     """Discrete inverse wavelet trafo between discrete L2 spaces."""
```

Comment:
trafo -> transform 

Also, there is no need to disallow other space exponents. Just need to filter that in the adjoint.